shared-bindings/alarm/__init__.c

```
@@ -0,0 +1,239 @@
+//|     is returned, and is also available as `alarm.wake_alarm`. Nothing is shut down
+//|     or interrupted. Power consumption will be reduced if possible.
+//|
+//|     If no alarms are specified, return immediately.
+//|     """
+//|     ...
+//|
+STATIC mp_obj_t alarm_wait_until_alarms(size_t n_args, const mp_obj_t *args) {
+    validate_objs_are_alarms(n_args, args);
+    common_hal_alarm_wait_until_alarms(n_args, args);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(alarm_wait_until_alarms_obj, 1, MP_OBJ_FUN_ARGS_MAX, alarm_wait_until_alarms);
+
+//| def sleep_until_alarms(*alarms: Alarm) -> Alarm:
+//|     """Go into a light sleep until awakened one of the alarms. The alarm causing the wake-up
+//|     is returned, and is also available as `alarm.wake_alarm`.
+//|
+//|     Some functionality may be shut down during sleep. On ESP32-S2, WiFi is turned off,
+//|     and existing connections are broken.
+//|
+//|     If no alarms are specified, return immediately.
+//|
+//|     **If CircuitPython is connected to a host computer,** `alarm.sleep_until_alarms()`
+//|     **does not go into light sleep.**
+//|     Instead, light sleep is simulated by doing `alarm.wait_until_alarms()`,
+//|     This allows the user to interrupt an existing program with ctrl-C,
+//|     and to edit the files in CIRCUITPY, which would not be possible in true light sleep
+//|     """
+//|     ...
+//|
+STATIC mp_obj_t alarm_sleep_until_alarms(size_t n_args, const mp_obj_t *args) {
+    validate_objs_are_alarms(n_args, args);
+    common_hal_alarm_sleep_until_alarms(n_args, args);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(alarm_sleep_until_alarms_obj, 1, MP_OBJ_FUN_ARGS_MAX, alarm_sleep_until_alarms);
+
+//| def exit_and_deep_sleep_until_alarms(*alarms: Alarm) -> None:
+//|     """Exit the program and go into a deep sleep, until awakened by one of the alarms.
+//|     This function does not return.
```

I think this should raise an exception like sys.exit and reload do. That way it can all be managed by main.c.

---

(see above).

I realized today that I need to make sure the filesystem caches are all flushed, etc., so doing it in `main.c` seems right.